Lib/test/test_pathlib.py

```diff
@@ -2913,6 +2914,20 @@ def test_passing_kwargs_deprecated(self):
         with self.assertWarns(DeprecationWarning):
             self.cls(foo="bar")
 
+    def test_from_uri_common(self):
+        P = self.cls
+        self.assertEqual(P.from_uri('file:foo/bar'), P('foo/bar'))
```

Comment:
Isn't this incorrect behaviour?

```pycon
# python3.11
>>> import pathlib
>>> pathlib.Path("foo/bar").as_uri()
ValueError: relative path can't be expressed as a file URI
```

Reply:
... but the robustness principle gets it share of criticism too. My mind isn't set!

Another thing worth considering: all other pathlib constructors accept all string inputs without raising exceptions. We've previously discussed and decided against performing validation, e.g. checking for control characters in Windows paths. Whether or not that logic extends here is another matter.

Reply:
Oh, I don't have a strong preference either way. It's more that: whatever pathlib decides to do here will influence what universal_pathlib will support. Like in the [workflow xkcd](https://xkcd.com/1172/), someone will rely on the behaviour 😄 

In support for relative file uris: people seem to expect it to work (see here https://github.com/fsspec/universal_pathlib/commit/662c2943deff680423e048f62326cfdb8405ef09#r123906140 for example).